user_sync/rules.py

```diff
@@ -291,24 +312,35 @@ def is_selected_user_key(self, user_key):
         return True
     
     def process_orphaned_dashboard_users(self):
-        remove_user_key_list = self.remove_user_key_list
+        remove_or_delete_user_key_list = self.remove_or_delete_user_key_list
             
         options = self.options
         remove_list_output_path = options['remove_list_output_path']
         remove_nonexistent_users = options['remove_nonexistent_users']
+        delete_list_output_path = options['delete_list_output_path']
+        delete_nonexistent_users = options['delete_nonexistent_users']
         
         max_deletions_per_run = options['max_deletions_per_run']
         max_missing_users = options['max_missing_users']
 
-        orphaned_dashboard_users = list(self.iter_orphaned_dashboard_users(self.options['managed_identity_types']))
-        self.logger.info('Orphaned users to be removed: %s', [self.get_dashboard_user_key(dashboard_user) for dashboard_user in orphaned_dashboard_users])
+        if self.need_to_process_orphaned_dashboard_users_for_remove:
+            orphaned_dashboard_users = list(self.iter_orphaned_dashboard_users(self.options['managed_identity_types']))
+            self.logger.info('Orphaned users to be removed: %s', [self.get_dashboard_user_key(dashboard_user) for dashboard_user in orphaned_dashboard_users])
+        elif self.need_to_process_orphaned_dashboard_users_for_delete:
+            orphaned_dashboard_users = list(self.iter_orphaned_dashboard_users(self.options['all_identity_types']))
+            self.logger.info('Orphaned users to be deleted: %s', [self.get_dashboard_user_key(dashboard_user) for dashboard_user in orphaned_dashboard_users])
+        else:
+            raise user_sync.error.AssertionException('User removal operation type invalid.')            
         
         number_of_orphaned_dashboard_users = len(orphaned_dashboard_users)
 
-        if (remove_list_output_path != None):
+        if remove_list_output_path != None:
```

Comment:
Great that you removed the parens, but...
- You should never say != None, the correct Python is "is not None"
- You shouldn't ever test for is not None when you are really asking "does it have a value", which is what you are asking in this case, you should just say:
```py
if remove_list_output_path:
```